lib/device-log/ios-log.js

```diff
@@ -25,20 +26,74 @@ class IOSLog {
-    } catch (err) {
-      logger.warn('Could not capture device log using libimobiledevice idevicesyslog. ' +
-                  'Libimobiledevice is probably not installed');
+    let cmd, args, env;
+    if ((this.realDeviceLogger || '').indexOf('idevicesyslog') !== -1) {
+      logger.debug('Attempting iOS device log capture via libimobiledevice idevicesyslog');
+      if (this.realDeviceLogger.toLowerCase() === 'idevicesyslog') {
+        cmd = 'idevicesyslog';
+        try {
+          // make sure it is available on the PATH
+          await fs.which('idevicesyslog');
+        } catch (err) {
+          throw new Error(`Unable to find system idevicesyslog: ${err.message}`);
+        }
+      } else {
+        // make sure the executable exists
+        if (!await fs.exists(this.realDeviceLogger)) {
+          throw new Error(`Unable to find idevicesyslog from 'realDeviceLogger' capability '${this.realDeviceLogger}'`);
+        }
+        cmd = this.realDeviceLogger;
+      }
+
+      args = ['-u', this.udid];
+      env = process.env;
+    } else if ((this.realDeviceLogger || '').indexOf('deviceconsole') !== -1) {
       logger.debug('Attempting iOS device log capture via deviceconsole');
-      spawnEnv.PATH = `${process.env.PATH}:${DEVICE_CONSOLE_PATH}`;
-      spawnEnv.DYLD_LIBRARY_PATH = `${DEVICE_CONSOLE_PATH}:${process.env.DYLD_LIBRARY_PATH}`;
-      this.proc = new SubProcess('deviceconsole', ['-u', this.udid], {env: spawnEnv});
+      let deviceconsole;
+      if (this.realDeviceLogger.toLowerCase() === 'deviceconsole') {
+        deviceconsole = DEVICE_CONSOLE_PATH;
+      } else {
+        // make sure that we have the path to the directory,
+        // not the actual executable
+        let stat;
+        try {
+          stat = await fs.stat(this.realDeviceLogger);
+        } catch (err) {
+          throw new Error(`Unable to find deviceconsole from 'realDeviceLogger' capability '${this.realDeviceLogger}': ${err.message}`);
+        }
+        if (stat.isDirectory()) {
+          deviceconsole = this.realDeviceLogger;
+        } else {
+          // make sure they've passed in `deviceconsole` and not something random
+          if (this.realDeviceLogger.lastIndexOf('deviceconsole') !== this.realDeviceLogger.length - 'deviceconsole'.length) {
+            throw new Error(`Unable to parse 'deviceconsole' installation directory from '${this.realDeviceLogger}'`);
+          }
+          // remove the executable, and trailing `/`, to get the install directory
+          deviceconsole = this.realDeviceLogger.substring(0, this.realDeviceLogger.length - 'deviceconsole'.length - 1);
+        }
+      }
+
+      logger.debug(`Using 'deviceconsole' from '${deviceconsole}'`);
+
+      cmd = 'deviceconsole';
+      args = ['-u', this.udid];
+
+      // set up the environment to be able to run deviceconsole
+      env = _.clone(process.env);
+      env.PATH = `${deviceconsole}:${process.env.PATH}`;
```

Comment:
rather than updating PATH and then calling the command relatively, why not just call the command using its absolute path (which we now know)?